esmf_regrid/schemes.py

```diff
@@ -575,29 +1051,186 @@ def __call__(self, cube):
             area-weighted regridding via :mod:`esmpy` generated weights.
 
         """
-        src_x, src_y = (_get_coord(cube, "x"), _get_coord(cube, "y"))
-
-        # Check the source grid matches that used in initialisation
-        if self.src_grid != (src_x, src_y):
-            raise ValueError(
-                "The given cube is not defined on the same "
-                "source grid as this regridder."
-            )
+        if cube.mesh is not None:
+            # src_mesh = cube.mesh
+            # location = cube.location
+            # if self.src != (src_mesh, location):
+            #     raise ValueError(
+            #         "The given cube is not defined on the same "
+            #         "source mesh as this regridder."
+            #     )
```

Comment:
How come? `Mesh` equality actually looks at `Mesh` contents ([code](https://github.com/SciTools/iris/blob/90e2c1dc51d4b70a739b8c1933b07e4f6185db23/lib/iris/experimental/ugrid/mesh.py#L907-L917)), rather than just the ID like it used to, so it ought to be able to cope with two identical `Mesh`es that have different IDs.

Reply:
I think the remaining problems were due to the fact that it was possible for masked data to appear in the connectivity of a mesh (for meshes of mixed type). There's probably a case to be made that these sorts of meshes are rare enough to ignore since I don't think I've encountered one in the wild. Either way, this is probably something to look at in another issue.

Reply:
see https://github.com/SciTools-incubator/iris-esmf-regrid/pull/198/commits/789c64757ab3df6b2171dd1036ae3c162dfc8b84